examples/kraken-websocket.yml

```diff
@@ -0,0 +1,405 @@
+asyncapi: 3.0.0
+
+info:
+  title: Kraken Websockets API
+  version: '1.8.0'
+  description: |
+    WebSockets API offers real-time market data updates. WebSockets is a bidirectional protocol offering fastest real-time data, helping you build real-time applications. The public message types presented below do not require authentication. Private-data messages can be subscribed on a separate authenticated endpoint. 
+
+    ### General Considerations
+
+    - TLS with SNI (Server Name Indication) is required in order to establish a Kraken WebSockets API connection. See Cloudflare's [What is SNI?](https://www.cloudflare.com/learning/ssl/what-is-sni/) guide for more details.
+    - All messages sent and received via WebSockets are encoded in JSON format
+    - All decimal fields (including timestamps) are quoted to preserve precision.
+    - Timestamps should not be considered unique and not be considered as aliases for transaction IDs. Also, the granularity of timestamps is not representative of transaction rates.
+    - At least one private message should be subscribed to keep the authenticated client connection open.
+    - Please use REST API endpoint [AssetPairs](https://www.kraken.com/features/api#get-tradable-pairs) to fetch the list of pairs which can be subscribed via WebSockets API. For example, field 'wsname' gives the supported pairs name which can be used to subscribe.
+    - Cloudflare imposes a connection/re-connection rate limit (per IP address) of approximately 150 attempts per rolling 10 minutes. If this is exceeded, the IP is banned for 10 minutes.
+    - Recommended reconnection behaviour is to (1) attempt reconnection instantly up to a handful of times if the websocket is dropped randomly during normal operation but (2) after maintenance or extended downtime, attempt to reconnect no more quickly than once every 5 seconds. There is no advantage to reconnecting more rapidly after maintenance during cancel_only mode.
+
+
+channels:
+  ping:
+    address: /
+    messages:
+      ping:
+        $ref: '#/components/messages/ping'
+  pong:
+    address: /
+    messages:
+      pong:
+        $ref: '#/components/messages/pong'
+
+  heartbeat:
+    address: /
+    messages:
+      heartbeat:
+        $ref: '#/components/messages/heartbeat'
+      
+  systemStatus:
+    address: /
+    messages:
+      systemStatus:
+        $ref: '#/components/messages/systemStatus'
+      
+  subscriptionStatus:
+    address: null
+    messages:
+      subscriptionStatus:
+        $ref: '#/components/messages/subscriptionStatus'
+  subscribe:
+    address: /
+    messages:
+      subscribe:
+        $ref: '#/components/messages/subscribe'
+  unsubscribe:
+    address: /
+    messages:
+      unsubscribe:
+        $ref: '#/components/messages/unsubscribe'
+
+      
+operations:
+  pingPong:
+    action: send
+    channel: 
+      $ref: '#/channels/ping'
+    reply:
+      channel: 
+        $ref: '#/channels/pong'
```

Comment:
What does the mirrored operation look like for the application that responds to the ping message? Would that be like so? 

```yaml
operations:
  pingRespond
    action: receive
    channel: 
      $ref: '#/channels/ping'
    reply:
      channel: 
        $ref: '#/channels/pong'
```
`receive` and `reply` seems a bit weird together 🤔

Reply:
Exactly like your example. That would make perfect sense for me.

Reply:
That makes perfect sense to me too. You're reacting to an incoming message, therefore sending another one.